subprojects/core/src/main/java/org/gradle/api/internal/artifacts/JavaEcosystemSupport.java

```diff
@@ -35,6 +37,31 @@
 
 public abstract class JavaEcosystemSupport {
     public static void configureSchema(AttributesSchema attributesSchema, final ObjectFactory objectFactory) {
+        configureUsage(attributesSchema, objectFactory);
+        configureBundling(attributesSchema);
+        configureTargetPlatform(attributesSchema);
+    }
+
+    private static void configureTargetPlatform(AttributesSchema attributesSchema) {
+        AttributeMatchingStrategy<Integer> targetPlatformSchema = attributesSchema.attribute(TargetJavaPlatform.MINIMAL_TARGET_PLATFORM_ATTRIBUTE);
+        targetPlatformSchema.getCompatibilityRules().add(TargetPlatformCompatibilityRules.class);
+        targetPlatformSchema.getDisambiguationRules().add(TargetPlatformDisambiguationRules.class, new Action<ActionConfiguration>() {
+            @Override
+            public void execute(ActionConfiguration config) {
+                // by default we will reject any producer which version is higher than
+                // the current Gradle runtime (if the consumer says nothing)
+                config.params(Integer.valueOf(JavaVersion.current().getMajorVersion()));
```

Comment:
I'm not sure I'd apply this rule. We don't know what the consumer is going to do with the incoming things at this level. Instead, I'd leave it unspecified and leave it to the plugins to set up the correct values on the configurations that they know about.

Reply:
Makes sense, I'm going to remove it. It also has the advantage of simplifying the compatibility rules.